zipkin-server/src/main/java/zipkin2/server/internal/elasticsearch/DynamicCredentialsFileLoader.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2015-2020 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin2.server.internal.elasticsearch;
+
+import com.google.common.util.concurrent.RateLimiter;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.Properties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.scheduling.annotation.Scheduled;
+
+import static zipkin2.server.internal.elasticsearch.ZipkinElasticsearchStorageConfiguration.PASSWORD_PROP;
+import static zipkin2.server.internal.elasticsearch.ZipkinElasticsearchStorageConfiguration.USERNAME_PROP;
+
+/**
+ * Loads username/password from credentials file.
+ *
+ * <p><em>NOTE:</em> This implementation currently uses blocking I/O loop without poll events.
+ * This means that spans will drop and api failures will occur for any time remaining in the refresh
+ * interval. A future version can tighten this by also using poll events.
+ */
+class DynamicCredentialsFileLoader {
+  static final Logger LOGGER = LoggerFactory.getLogger(DynamicCredentialsFileLoader.class);
+  static final String CREDENTIALS_REFRESH_INTERVAL =
+    "zipkin.storage.elasticsearch.credentials-refresh-interval";
+
+  private final String credentialsFile;
+
+  private final BasicCredentials basicCredentials;
+
+  // Log an exception every 10 seconds.
+  private final RateLimiter rateLimiter = RateLimiter.create(0.1);
+
+  public DynamicCredentialsFileLoader(BasicCredentials basicCredentials,
+    String credentialsFile) {
+    this.basicCredentials = basicCredentials;
+    this.credentialsFile = credentialsFile;
+  }
+
+  @Scheduled(fixedRateString = "${" + CREDENTIALS_REFRESH_INTERVAL + "}")
```

Comment:
actually if we use a normal single thread scheduled executor and make this type closeable, it should be less error prone on things like this. Spring automatically closes beans on shutdown.

Also, this as a bean allows something to synchronously call load auth fail. If we don't do this now, move it to a comment.

Reply:
added this note in the type doc